BlazarUI/app/scripts/components/shared/PageHeader.jsx

```diff
@@ -1,10 +1,38 @@
 import React, {Component, PropTypes} from 'react';
 
-class PageHeader extends Component{
+class PageHeader extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      headerClass: ''
+    };
+  }
+
+  componentDidMount() {
+    window.onscroll = this.handleScroll.bind(this);
+  }
+
+  componentWillUnmount() {
+    window.onscroll = null;
+  }
+
+  handleScroll(event) {
+    let position = window.pageYOffset;
+    if (position > 10) {
```

Comment:
there's a lot of unnecessary calls to set state and render here. add a state property that keeps track of if we want the page header to be small, then you can do something like `if (position > 10 && !this.state.isPageHeaderSmall)`.